lib/components/RunFrame/RunFrame.tsx

```diff
@@ -209,9 +210,20 @@ export const RunFrame = (props: RunFrameProps) => {
         )
   const lastFsMapRef = useRef<Map<string, string> | null>(null)
   const lastEntrypointRef = useRef<string | null>(null)
+  const { isCircuitJsonFile } = useCircuitJsonFile({
+    mainComponentPath: props.mainComponentPath,
+    fsMap,
+    isLoadingFiles: props.isLoadingFiles,
+    onCircuitJsonChange: props.onCircuitJsonChange,
+    setCircuitJson,
+    setError,
```

Comment:
Never pass setters as props

Im also pretty sure you shouldnt have onCircuitJsonChange

otherwise pretty decent design